src/uuid-encoding/Base32.cs

```diff
@@ -82,6 +82,14 @@ public static class Base32 {
                     bitsLeft -= 8;
                 }
             }
+
+            // Ramaing bits lead to ambiguity, since different input strings lead to the same 
+            // result when ignoring those bits) 
+            if ((buffer & ((1 << bitsLeft) - 1)) != 0)
+            {
+                throw new DecodingException("Remaining bits: " + bitsLeft);
+            }
+
             // We'll ignore leftover bits for now.
```

Comment:
We should figure out about the next couple of lines which are commented-out - Is it the equivalent check? Does it check something else? Can we remove it?

Reply:
The conditions are not quite equivalent, the second condition refers to the case from the removed test case (16 7s vs 17 7s), it catches only the cases when the remaining bits are more than the shift size (5 in our case). If we do not validate the length of the input before decoding - it'd make sense to out comment the second condition as well:

```
> Input: aaaaaaaaaaaaaaaaaaaaaaaaaa
> Output: 00000000-0000-0000-0000-000000000000
> Input: aaaaaaaaaaaaaaaaaaaaaaaaab
> Remaining bits (cond #1): 2
> Output: 00000000-0000-0000-0000-000000000000
> Input: aaaaaaaaaaaaaaaaaaaaaaaaac
> Remaining bits (cond #1): 2
> Output: 00000000-0000-0000-0000-000000000000
> Input: aaaaaaaaaaaaaaaaaaaaaaaaad
> Remaining bits (cond #1): 2
> Output: 00000000-0000-0000-0000-000000000000
> Input: aaaaaaaaaaaaaaaaaaaaaaaaaaa
> Bits left (cond #2): 7
> Output: 00000000-0000-0000-0000-000000000000
> Input: aaaaaaaaaaaaaaaaaaaaaaaaae
> Output: 00000000-0000-0000-0000-000000000001

```